src/test/unit_tests/test_bucketspace_versioning.js

```diff
@@ -1907,6 +1965,36 @@ mocha.describe('bucketspace namespace_fs - versioning', function() {
             await fs_utils.file_must_not_exist(path.join(full_multi_delete_path, key1));
             await delete_object_versions(full_multi_delete_path, key1);
         });
+
+        mocha.it('delete multiple objects - versioning enabled (more than 1 level) ', async function() {
+            const self = this; // eslint-disable-line no-invalid-this
+            self.timeout(150000);
+            const nested_key_level4 = '/company/department/team/2024/January/employee_123.txt';
+            const arr_keys_only = [];
+            const arr_keys_and_version_id = [];
+            const num_of_versions = 5;
+            for (let i = 0; i < num_of_versions; i++) {
+                const body_to_add = `zzzzz-${i}`;
+                const res = await account_with_access.putObject({
+                    Bucket: delete_multi_object_test_bucket, Key: nested_key_level4, Body: body_to_add });
+                arr_keys_only.push({ Key: nested_key_level4 });
+                arr_keys_and_version_id.push({ Key: nested_key_level4, VersionId: res.VersionId });
+            }
+
+            // no version-id (latest)
+            const delete_res = await account_with_access.deleteObjects({
+                    Bucket: delete_multi_object_test_bucket, Delete: { Objects: arr_keys_only } });
+            assert.equal(delete_res.Deleted.length, 1);
+
+            // with version-id
+            const delete_res2 = await account_with_access.deleteObjects({
+                Bucket: delete_multi_object_test_bucket, Delete: { Objects: arr_keys_and_version_id } });
+            assert.equal(delete_res2.Deleted.length, num_of_versions);
+
+            // delete the delete marker of latest version
+            await s3_uid6.deleteObject({ Bucket: delete_multi_object_test_bucket,
+                Key: delete_res.Deleted[0].Key, VersionId: delete_res.Deleted[0].DeleteMarkerVersionId});
+        });
```

Comment:
I would add here a check of the fs directly using list object versions, so we can validate that the object was deleted completely not only through the responses

Reply:
Added, thanks.